client/security/src/main/java/org/eclipse/kapua/client/security/bean/KapuaPrincipalImpl.java

```diff
@@ -128,13 +125,6 @@ public boolean equals(Object obj) {
             return false;
         }
         KapuaPrincipalImpl other = (KapuaPrincipalImpl) obj;
-        if (accountId == null) {
-            if (other.accountId != null) {
-                return false;
-            }
-        } else if (!accountId.equals(other.accountId)) {
```

Comment:
Wouldn't it be possible to have the same username and clientId in multiple accounts?

Reply:
no because the username is unique in the platform (it's not possible to have the same username in 2, or more, different acocunt).
If a user connects with multiple clients each client should have different clientId (otherwise the broker disconnects the first client connected)

Reply:
Ah okay, I didn't realize that the username is unique, but I was able to confirm this.  In that case, I agree that just comparing the principal name would seem sufficient.